java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/ui/VSCodeURLDisplayer.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.lsp.server.ui;
+
+import java.net.URL;
+import org.netbeans.modules.java.lsp.server.protocol.HtmlPageParams;
+import org.openide.awt.HtmlBrowser.URLDisplayer;
+
+/**
+ *
+ * @author Jan Horvath
+ */
+public class VSCodeURLDisplayer extends URLDisplayer {
```

Comment:
How is this instance registered?

The preferred way of registering these services is to create `AbstractLspURLDisplayer` with all the implementation, but without registering it. Then Create a subclass in `nbcode/integration` module with `@ServiceProvider` registration.

Make the class [abstract with protected constructor](https://github.com/apache/netbeans/blob/36443ae276eee4a2b03fd59ce28af02b576e53fd/java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/ui/AbstractLspStatusDisplayer.java#L29).